### DISCUSSION TITLE
Adding table copies with ejabberd_mnesia

### DISCUSSION THREAD

**Discussion by slezakattack** [processone/ejabberd#3603] [Questions] [answered]
Hello,

I have a table defined in my code as:
```
TableOpts = [{ram_copies, [node()]}, {attributes, record_info(fields, my_table)}],
ejabberd_mnesia:create(?MODULE, my_table, TableOpts).
```
My ejabberd cluster is in kubernetes and changes are deployed using rolling restarts. I've learned that if a new mnesia table is introduced with `ejabberd_mnesia:create/3`, only one node possesses the table contents. My expectation would be that `ejabberd_mnesia` would determine the table exists and instead perform an `mnesia:add_table_copy(my_table, node(), ram_copies)`. Is this considered a bug? If so, I can create a PR but wanted to know if this the preferred direction or if I just need to explicitly call `mnesia:add_table_copy/3` in my code. Thanks!

**Comment by badlop** [comment, marked as answer]
How do other modules handle this? Does it work for them, and not for your one? What's the difference between the other modules and your one, that could explain the different behaviour?

**Reply by slezakattack** [reply]
I apologize, I just came across this documentation https://docs.ejabberd.im/admin/upgrade/#soft-upgrade-process so it seems if we follow the "leave_cluster" and "join_cluster" commands, then the mnesia table copies are added correctly.
